### ISSUE TITLE
S3 on Frankfurt: ERROR: ErrorInvalidRequest - Please use AWS4-HMAC-SHA256

### ISSUE THREAD

**Issue by zdco** [bji/libs3#15]
Hi,
  When using libs3 with S3 on Frankfurt (eu-central-1), I get the following error:

ERROR: ErrorInvalidRequest 
Message: The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256. 
Extra Details: 
   RequestId: xxxxxx 
   HostId: xxxxxxx

Has anybody ran into this? Any idea how to fix it?

**Comment by barzoj** [comment]
Hi,

I think AWS4-HMAC-SHA256 is not supported by libs3. To workaround this problem use one of older S3 regions. You may also use AWS SDK for C++

https://aws.amazon.com/ru/blogs/aws/introducing-the-aws-sdk-for-c/

but it lacks support of older (v2) authentification mechanism

**Comment by bjitivo** [comment]
Yes, please us the official Amazon SDK if your project is C++ based!  libs3 has not had active development for some time now unfortunately.

Although looking at the example C++ code provided on that page ... that is definitely not a dialect of C++ that I would ever want to write code in.  But that's a personal preference I suppose.
